### ISSUE TITLE
Identify where the change comes from

### ISSUE THREAD

**Issue by ed-holland** [larshp/abapGit#3546]
When a change is commited, is it possible to identify if a changes is made via ADT or via visual code?

Reason why i am asking is that if the code is not checked in via ADT, then it needs to be loaded into SAP to make sure that all stays consistent. If it can be detected, then this pull can be automated.

Related Question: Can the pull be done for just one specific commit?

**Comment by larshp** [comment]
disregarding editor, it is possible for all developers to make errors, example of active program below

for abapGit development itself, the changes are proposed via pull requests which are checked by abaplint, which provides a small assurance that the ABAP code is consistent.  Consistent changes are then merged to master

![image](https://user-images.githubusercontent.com/5888506/85838261-966fa580-b799-11ea-8933-bf2cdf8df100.png)

**Comment by mbtools** [comment]
there's only so much one can do on the development side. syntax checks, linting, and unit tests - all can be integrated in AG - should catch most inconsistencies in your dev box, no matter where the code came from. 

if you really must know, logging the HTTP requests into dev might give you some indication.

**Comment by ed-holland** [comment]
Let me rephrase the question.

When an Abap is not changed via ADT (where AbapGit resides) but via other means, then a complete syntax check can only be done by importing the change into the AbapGit system. If a pipeline/workflow can determine this, then the import / pull can be automated.

**Comment by mbtools** [comment]
I would run ATC as a nightly job to check things independent of abapgit.

**Comment by ed-holland** [comment]
thanks for the help. Closing
